src/com/dotcms/integritycheckers/ContentFileAssetIntegrityChecker.java

```diff
@@ -103,7 +103,12 @@ public void executeFix(final String endpointId) throws DotDataException, DotSecu
 
             int counter = versionCount.get(oldIdentifier);
             boolean isTheLastConflict = counter == 1 ? true : false;
-            fixContentletConflicts(result, Structure.STRUCTURE_TYPE_FILEASSET, isTheLastConflict);
+            try{
+            	fixContentletConflicts(result, Structure.STRUCTURE_TYPE_FILEASSET, isTheLastConflict);
+            }
+            catch(IOException e){
+            	throw new DotDataException(e.getMessage(),e);
```

Comment:
Since what we are doing is re-throwing as a DotDataException, wouldn't be a good alternative to do this here https://github.com/dotCMS/core/pull/9199/commits/ae081bda4c9ec3757cfee68e3b8737d81247e2a7#diff-e7ef0931d4d3ce042811c214b568916cR560 instead. Surround that line with try catch DotDataException and rethrow. That would prevent changing the signature of  generateNewContentlet, moveInodeFolder and fixContentletConflicts and would have the same result.